pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -722,3 +722,28 @@ func buildOutboundCatchAllNetworkFilterChains(_ *ConfigGeneratorImpl,
 	chains = append(chains, &listener.FilterChain{Name: VirtualOutboundCatchAllTCPFilterChainName, Filters: filterStack})
 	return chains
 }
+
+// EvaluateListenerFilterPredicates runs through the ListenerFilterChainMatchPredicate logic
+// This is exposed for testing only, and should not be used in XDS generation code
+func EvaluateListenerFilterPredicates(predicate *listener.ListenerFilterChainMatchPredicate, invertMatch bool, port int) bool {
```

Comment:
If this is only for test, can you move it out to the test file?

Reply:
No, it's shared by multiple packages

Reply:
oh I guess we can move it to the xdstest package though

Reply:
That's good

Reply:
moved it